content/docs/ops/runbook/rotating-cloudfoundry-diego.md

```diff
@@ -0,0 +1,206 @@
+deployments are focused on the Consul components for Cloud Foundry and Diego.
+
+Follow the following documentation to get familiar with which values are going
+to be replaced.
+
+- [Cloud Foundry AWS Stub](http://docs.cloudfoundry.org/deploying/aws/cf-stub.html)
+- [Security Configuration for Consul](http://docs.cloudfoundry.org/deploying/common/consul-security.html)
+- [Generating all the Cloud Foundry certificates](https://github.com/18F/cg-deploy-cf#how-to-generate-the-final-manifest)
+
+## Prepare a local directory for rotating secret files
+
+Create a directory you can use to generate all the various artifacts that need
+to be rotated for the secrets files. Create a `./tmp` directory inside the
+`cg-deploy-cf` repository. Delete this directory at the end of secrets rotation.
+
+## Dealing with multiple secret files
+
+The Cloud Foundry manifest is split into two parts, a `main.yml` and
+`external.yml`. Internal secrets can be rotated either all at once or on a
+rolling schedule of deployments. You can pull the information needed to decrypt
+these from the `cg-deploy-cf` pipeline. You can easily grab the information
+needed to [decrypt and encrypt secrets]({{< relref "docs/ops/secrets.md" >}}) from
+Concourse using the `fly-cli`.
+
+```sh
+# note the target name `fr` here is the output of `fly targets`
+$ fly -t fr get-pipeline --pipeline deploy-cf | \
+  spurce json | \
+  jq -r '.resources | select(.name | test("common-"))'
+```
+
+### Internal secrets (main.yml)
+
+1. Rotate all UAA user passwords with passwords matching the UAA restrictions
+   set in [the spec](https://github.com/18F/uaa-customized-boshrelease/) for our
+   `uaa-customized-boshrelease`.
+    - `uaa.password.policy.minLength`
+    - `uaa.password.policy.requireUpperCaseCharacter`
+    - `uaa.password.policy.requireLowerCaseCharacter`
+    - `uaa.password.policy.requireDigit`
+    - `uaa.password.policy.requireSpecialCharacter`
+1. Rotate all the `uaa.clients.*` secrets.
+    - For `uaa.clients.doppler` change `loggregator.uaa.client_secret` as well.
+    - For `uaa.clients.tcp_emitter` change `acceptance_tests.oauth_password` as well.
+
+#### Generate all the certificates
+
+Add all the CA certificates signed by the Bosh root CA certificate. These root
+certificate and key files can be grabbed with the `--grab-cert` flag for the
+`generate-all-certificates.sh` file in
+[18F/cg-deploy-cf](https://github.com/18F/cg-deploy-cf).
+
+The `ci_env` environmental variable is the value for the name of the Concourse
+you would like to target to get the root CA certificate from. (e.g. `fly targets`).
+From inside the `./tmp` directory, run the following commands:
+
+```sh
+$ ci_env=fr ../generate-all-certificates.sh --grab-cert && \
+  mv all-cf-certs staging-cf-diego-certs
+$ ci_env=fr ../generate-all-certificates.sh --grab-cert && \
+  mv all-cert-certs production-cf-diego-certs
+```
+
+This script will generate all 25 of the certificates for both Diego and
```

Comment:
Let's not hard-code certificate counts--the number doesn't really matter and will probably change over time.